app/src/interfaces/project-interfaces.ts

```diff
@@ -38,6 +26,19 @@ export interface IProject {
   regions_name_list: string;
 }
 
+/**
+ * An interface representing the project table.
+ *
+ * @export
+ * @interface IProject
+ */
+export interface IProjectWithDetails {
+  id: number;
+  project: IProjectDetailsForm;
+  objectives: IProjectObjectivesForm;
+  location: IProjectLocationForm;
```

Comment:
I think there are some tweaks we can make here. The general structure is spot on, but it will need to use different interfaces (not the Form ones). 

The Form interfaces only account for the fields the user interacts with, but in our response from the API we'll have extra fields like the primary key and revision_count for each row, as well as any additional details that are generated by the backend (if any). 

That being said, if its working now, then lets not worry about it in this PR so that we don't hold up progress on the rest of the dependent tickets.  We'll have to do some refinement inevitably anyways as we start putting more of the view pages together, etc.

Reply:
My intention with using the form interfaces was to be ready for the upcoming sprints where we need to edits each of these sections individually.  Not sure what the plan is at this point for those, but it would be good to re-use the form interfaces?

Reply:
Gotcha, that makes sense.  In that case, we can still make new interfaces, but we can extend the form interfaces to ensure that those fields are present, but still allow us to include extras